cmd/srcd/cmd/sql_test.go

```diff
@@ -0,0 +1,209 @@
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+
+	_, err = s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	buf, err := s.RunSQL(context.TODO(), "select * from repositories")
+	require.NoError(err)
+
+	expected := `+---------------+
+| REPOSITORY ID |
++---------------+
+| reponame      |
++---------------+
+`
+	require.Contains(buf.String(), expected)
+}
+
+func (s *SQLTestSuite) TestNoInit() {
+	require := s.Require()
+
+	buf, err := s.RunSQL(context.TODO(), "show tables")
+	require.NoError(err)
+
+	require.Contains(buf.String(), showTablesOutput)
+}
+
+func (s *SQLTestSuite) TestValidQueries() {
+	testCases := []string{
+		"show tables",
+		"SHOW TABLES",
+		"show tables;",
+		"SHOW TABLES;",
+		" show tables ; ",
+		" SHOW TABLES ; ",
+		"	show tables	;	",
+		"	SHOW TABLES	;	",
+		"/* comment */ show tables;",
+		`/* multi line
+			comment */
+			show tables;`,
+	}
+
+	for _, query := range testCases {
+		s.T().Run(query, func(t *testing.T) {
+			assert := assert.New(t)
+
+			buf, err := s.RunSQL(context.TODO(), query)
+			assert.NoError(err)
+
+			assert.Contains(buf.String(), showTablesOutput)
+		})
+	}
+}
+
+func (s *SQLTestSuite) TestWrongQuery() {
+	testCases := []struct {
+		query string
+		err   string
+	}{
+		{
+			query: "show",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
+		},
+		{
+			query: "show tables; show tables",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
+		},
+		{
+			query: "select from repositories",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
+		},
+		{
+			query: "select * from nope",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: table not found: nope",
+		},
+		{
+			query: "insert into repositories values ('myrepo')",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: table doesn't support INSERT INTO",
+		},
+		{
+			query: "select nope from repositories",
+			err:   `rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: column "nope" could not be found in any table in scope`,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.T().Run(tc.query, func(t *testing.T) {
+			assert := assert.New(t)
+
+			buf, err := s.RunSQL(context.TODO(), tc.query)
+			assert.Error(err)
+
+			assert.Contains(buf.String(), tc.err)
+		})
+	}
+}
+
+func (s *SQLTestSuite) TestREPL() {
+	require := s.Require()
+
+	var out, in bytes.Buffer
+
+	command := s.CommandContext(context.TODO(), "sql")
+	command.Stdout = &out
+	command.Stderr = &out
+	command.Stdin = &in
+
+	io.WriteString(&in, "show tables;\n")
+	io.WriteString(&in, "describe table repositories;\n")
+
+	err := command.Start()
+	require.NoError(err)
+
+	err = command.Wait()
```

Comment:
could you please help me to understand how it works.
`.Wait` should block until command exits. But `repl` doesn't exit by itself and I don't see where you send stop signal to it.

Reply:
Yes, this is tricky, I will add some comments to the code.

What I _think_ is happening is that the [readline lib](github.com/chzyer/readline) we use must detect that it is not in a terminal, and exits when sdtin is consumed.

You can see it if you run the bin like this. It runs the query, prints the result and exits.
```
$ echo "show tables;" | ./srcd sql
```

So we are still testing [the code that processes](https://github.com/src-d/engine/blob/master/cmd/srcd/cmd/sql.go#L108) the lines read by the repl, but without interaction. There are things we can't test, like exit on `exit;`, but at least this will fail if we break something major there.

Reply:
Added comments in 1a374ba, will squash it before merging